redbot/core/utils/__init__.py

```diff
@@ -498,3 +508,81 @@ def map(self, func: Callable[[_T], Union[_S, Awaitable[_S]]]) -> AsyncIter[_S]:
             raise TypeError("Mapping must be a callable.")
         self._map = func
         return self
+
+
+def get_end_user_statement(file: Union[Path, str]) -> Optional[str]:
+    """
+    This function attempts to get the ``end_user_data_statement`` key from a cogs ``info.json``. This will log the reason if ``None`` is returned.
+
+    Parameters
+    ----------
+    file: Union[pathlib.Path, str]
+        The ``__file__`` variable for the cogs ``__init__.py`` file.
+
+    Returns
+    -------
+    Optional[str]
+        The end user statement found in the info.json or ``None`` if there was an issue finding one.
+
+    Examples
+    --------
+    >>> # In cogs `__init__.py`
+    >>> from redbot.core.utils import get_end_user_statement
+    >>> __red_end_user_data_statement__  = get_end_user_statement(__file__)
+    >>> def setup(bot):
+            ...
+    """
+    try:
+        file = Path(file).parent.absolute()
+        info_json = file / "info.json"
+        statement = get_end_user_statement_or_raise(info_json)
+    except FileNotFoundError:
+        log.critical("'%s' does not exist.", str(info_json))
+    except KeyError:
+        log.critical("'%s' is missing an entry for 'end_user_data_statement'", str(info_json))
+    except json.JSONDecodeError as exc:
+        log.critical("'%s' is not a valid JSON file.", str(info_json), exc_info=exc)
+    except UnicodeError as exc:
+        log.critical("'%s' has a bad encoding.", str(info_json), exc_info=exc)
+    except Exception as exc:
+        log.critical(
+            "There was an error when trying to load the end user statement from '%s'.",
+            str(info_json),
+            exc_info=exc,
+        )
+    else:
+        return statement
+    return None
+
+
+def get_end_user_statement_or_raise(file: Union[Path, str]) -> str:
+    """
+    This function attempts to get the ``end_user_data_statement`` key from a cogs ``info.json``.
+
+    Parameters
+    ----------
+    file: Union[pathlib.Path, str]
+        The ``__file__`` variable for the cogs ``__init__.py`` file.
+
+    Returns
+    -------
+    str
+        The end user statement found in the info.json.
+
+    Raises
+    ------
+    FileNotFoundError
+        When ``info.json`` does not exist.
+    KeyError
+        When ``info.json`` does not have the ``end_user_data_statement`` key.
+    json.JSONDecodeError
+        When ``info.json`` can't be decoded with ``json.load()``
+    UnicodeError
+        When ``info.json`` can't be decoded due to bad encoding.
+    Exception
+        Any other exception raised from ``pathlib`` and ``json`` modules when attempting to parse the ``info.json`` for the ``end_user_data_statement`` key.
+    """
+    file = Path(file).parent.absolute()
+    info_json = file / "info.json"
+    with info_json.open() as fp:
```

Comment:
`info.json` files need to be encoded in utf-8 and it isn't always the default encoding (especially on Windows).
```suggestion
    with info_json.open(encoding="utf-8") as fp:
```